test/box-luatest/gh_6085_limit_iteration_in_space_test.lua

```diff
@@ -90,3 +90,39 @@ g.test_limit_replace = function()
         check_fiber_slice(endless_replace_func, timeout, delta)
     end)
 end
+
+g.before_test('test_limit_on_sigurg', function()
+    g.server = server:new({
+        alias = 'default',
+    })
+    g.server:start()
+end)
+
+g.test_limit_on_sigurg = function()
+    local pid = g.server:eval('return box.info.pid')
+    local os = require('os')
+    local clock = require('clock')
+
+    g.server:exec(function()
+        local fiber = require('fiber')
+
+        local s = box.schema.create_space('tester')
+        s:create_index('pk')
+        for i = 1, 1000 do
+            s:insert{i}
+        end
+        local timeout = 10
+        fiber.set_max_slice(timeout)
+    end)
+
+    local start_time = clock.monotonic()
+    local future = g.server.net_box:eval('while true do box.space.tester:select{} end', {}, {is_async=true})
+    -- Wait while fiber will be waken up.
+    future:wait_result(2)
```

Comment:
You should check that `wait_result` actually returned an error here. Also, please reduce the timing of the test - 10 second of busy looping is way too long.

Reply:
done

Reply:
> 10 second of busy looping is way too long
10 second in the case of failed test